The generated import goPackageName no longer honour the package name

The removed logic located in [here](https://github.com/micro/protoc-gen-micro/compare/8936c52435e55ac5590fb6f9a7e2ed374a561ae9...master#diff-338bea067d2a01b69c90f1c032c6a24bL309)

It used to generate the package name based on the package name defined in the target `.proto` file however it has been removed.

This is causing an issue when the package name conflict with other generated code.

In our case, one of the generated interface names has the same name as the import package name. After I rollback to `8936c5` everything work just fine.

Can we add that logic back to https://github.com/micro/protoc-gen-micro/blob/master/generator/generator.go#L527, please?

Hi, this is causing issues for us now.

We have existing code that imports protos that have the same go package name as the current service, and this breaks our current builds.

For a simple example of what I'm talking about, say we have two services `api.test` and `srv.test`. 
The apis proto imports the service's proto to use it's type.

We are now getting code generated like this:

```
func RegisterTestHandler(s server.Server, hdlr TestHandler, opts ...server.HandlerOption) error {
	type test interface {
		Test(ctx context.Context, in *test.TestRequest, out *test.TestResponse) error
	}
	type Test struct {
		test
	}
	h := &testHandler{hdlr}
	return s.Handle(s.NewHandler(&Test{h}, opts...))
}
```

This is causing issues because the `test` interface is conflicting with the `test` package name and  `test.TestRequest` / `test.TestResponse` in the interface won't resolve and causes a compilation error of:

```
proto/test/test.micro.go:80:37: test.TestRequest undefined (type test has no method TestRequest)
proto/test/test.micro.go:80:60: test.TestResponse undefined (type test has no method TestResponse)
```

May I humbly suggest that this may be readily resolved by changing the interface name to `testInterface` instead which avoids this conflict without having to mess with import names etc?

Something like this: https://github.com/micro/protoc-gen-micro/pull/43

--- a/generator/generator.go
+++ b/generator/generator.go
@@ -81,7 +81,7 @@ type Plugin interface {
 	Generate(file *FileDescriptor)
 	// GenerateImports produces the import declarations for this file.
 	// It is called after Generate.
-	GenerateImports(file *FileDescriptor)
+	GenerateImports(file *FileDescriptor, imports map[GoImportPath]GoPackageName)
 }
 
 var plugins []Plugin
@@ -1254,7 +1254,7 @@ func (g *Generator) generateImports() {
 	g.P()
 	// TODO: may need to worry about uniqueness across plugins
 	for _, p := range plugins {
-		p.GenerateImports(g.file)
+		p.GenerateImports(g.file, imports)
 		g.P()
 	}
 	g.P("// Reference imports to suppress errors if they are not otherwise used.")

--- a/plugin/micro/micro.go
+++ b/plugin/micro/micro.go
@@ -40,6 +40,7 @@ var (
 	contextPkg string
 	clientPkg  string
 	serverPkg  string
+	pkgImports map[generator.GoPackageName]bool
 )
 
 // Init initializes the plugin.
@@ -82,7 +83,7 @@ func (g *micro) Generate(file *generator.FileDescriptor) {
 }
 
 // GenerateImports generates the import declaration for this file.
-func (g *micro) GenerateImports(file *generator.FileDescriptor) {
+func (g *micro) GenerateImports(file *generator.FileDescriptor, imports map[generator.GoImportPath]generator.GoPackageName) {
 	if len(file.FileDescriptorProto.Service) == 0 {
 		return
 	}
@@ -92,6 +93,13 @@ func (g *micro) GenerateImports(file *generator.FileDescriptor) {
 	g.P(serverPkg, " ", strconv.Quote(path.Join(g.gen.ImportPrefix, serverPkgPath)))
 	g.P(")")
 	g.P()
+
+	// We need to keep track of imported packages to make sure we don't produce
+	// a name collision when generating types.
+	pkgImports = make(map[generator.GoPackageName]bool)
+	for _, name := range imports {
+		pkgImports[name] = true
+	}
 }
 
 // reservedClientName records whether a client name is reserved on the client side.
@@ -103,7 +111,11 @@ func unexport(s string) string {
 	if len(s) == 0 {
 		return ""
 	}
-	return strings.ToLower(s[:1]) + s[1:]
+	name := strings.ToLower(s[:1]) + s[1:]
+	if pkgImports[generator.GoPackageName(name)] {
+		return name + "_"
+	}
+	return name
 }
 
 // generateService generates all the code for the named service.

